### ISSUE TITLE
Move resin-data to external drive

### ISSUE THREAD

**Issue by bbqsrc** [balena-os/balena-raspberrypi#401]
Is this possible? I can find no documentation about it either way.

I have concerns about an SD card hosting entire Docker environments and have an M.2 enclosure just sitting there begging to be used. :smile:

**Comment by floion** [comment]
Hi. So just to clarify, you do not want to have an additional storage drive as described here https://forums.balena.io/t/best-practise-to-use-an-external-drive/21739 but rather replace the data partition from the sd card to be on external drive?

**Comment by bbqsrc** [comment]
Yeah. My logic is that it would simplify the updating process, as then I can just change SD cards to update the balena OS while my data remains persistent on a separate disk.

**Comment by floion** [comment]
Unfortunately that is not something we support at the moment. Will try and see if we can get that to some sort of brainstorm

**Comment by bbqsrc** [comment]
Cool. How does the current partition layout get generated? /etc/fstab is very empty. :smile:

**Comment by floion** [comment]
It's done at build time by this class https://github.com/balena-os/meta-balena/blob/development/meta-balena-common/classes/image_types_resin.bbclass

**Comment by bbqsrc** [comment]
Okay, how does the OS mount them? Blindly mounting by partition label?

**Comment by floion** [comment]
Here are the yocto recipes for doing that: https://github.com/balena-os/meta-balena/tree/development/meta-balena-common/recipes-support/resin-mounts

And the relevant systemd units are here https://github.com/balena-os/meta-balena/tree/development/meta-balena-common/recipes-support/resin-mounts/resin-mounts

**Comment by agherzan** [comment]
@bbqsrc you will probably find this helpful: https://github.com/balena-os/meta-balena/pull/1677

**Comment by nonnos** [comment]
No news regarding this? @bbqsrc what was your final solution?

**Comment by bbqsrc** [comment]
Ended up not being able to use balena at all. 🙂
